### PR TITLE
fsthttp: don't call Close() on abireq if we haven't written all the b…

### PR DESCRIPTION
**Pull request by dgryski**
…ytes

### CODE REVIEW

**Review by joeshaw — APPROVED** [review]
👍

File: fsthttp/request.go [hunk 1]
```diff
@@ -282,10 +282,15 @@ func (req *Request) Send(ctx context.Context, backend string) (*Response, error)
...
+			_, copyErr := io.Copy(abiReqBody, req.Body)
+			errc <- maybeWrap(copyErr, "copy body")
 			errc <- maybeWrap(req.Body.Close(), "close user body")
-			errc <- maybeWrap(abiReqBody.Close(), "close request body")
+			if copyErr != nil {
+				// If there was an error copying the body, we *don't* want to Close() the abi req.
+				// This tells the wasm server that the body is incomplete so it knows not to
+				// terminate the sent chunked body with a valid final chunk.
+				errc <- maybeWrap(abiReqBody.Close(), "close request body")
+			}
```
Comment (peterbourgon) [inline fsthttp/request.go:293]: Then shouldn't this be `if copyErr == nil`?

Comment (dgryski) [reply fsthttp/request.go:293]: Bah.  Yes.  Of course it should.

Comment (dgryski) [reply fsthttp/request.go:293]: https://github.com/fastly/compute-sdk-go/pull/20

**Pull request merged**